package/MDAnalysis/analysis/rdf.py

```diff
@@ -229,6 +234,12 @@ def __init__(self,
         self.rdf_settings = {'bins': nbins,
                              'range': range}
         self._exclusion_block = exclusion_block
+        assert exclude_same is None or exclude_same in ['residue', 'segment', 'chain'], (
```

Comment:
Better to throw an exception than assert

Reply:
Changed this